### PR TITLE
Add project command

### PR DESCRIPTION
**Pull request by liamchampton**
**Problem**
The clone command needs to be part of a parent command. It also needs to have less flag requirements.

**Solution**
1. Introduce a parent command - `project`. This enables a user to enter a git url and the extraction of the previous flags needed `owner` & `repo` are extracted automatically.
2. Example command - `./installer project -r https://github.com/microclimate-dev2ops/microclimateGoTemplate ./downloadtest

**Tested**
Bats test output:
```
➜  codewind-installer git:(add-project-cmd) ✗ bats integration.bats
 ✓ invoke install command - default to latest tag
 ✓ invoke start command - Start dockerhub images'
 ✓ invoke stop-all command - Stop dockerhub images'
 ✓ invoke remove command - remove dockerhub images
 ✓ invoke project command - using -r download microclimate-dev2ops/microclimateGoTemplate into ./downloadtest

5 tests, 0 failures
```
Signed-off-by: Liam Hampton liam.hampton@ibm.com

### CODE REVIEW

File: actions/project.go [hunk 1]
```diff
@@ -22,10 +22,25 @@ import (
...
+func DownloadTemplate(c *cli.Context) {
+	destination := c.Args().Get(0)
+
+	if destination == "" {
+		log.Fatal("destination not set")
+	}
+
+	repoURL := c.String("r")
+
+	// expecting string in format 'https://github.com/<owner>/<repo>
```
Comment (jcockbain96) [inline actions/project.go]: Missing a closing '?

File: actions/commands.go [hunk 2]
```diff
@@ -35,33 +35,20 @@ func Commands() {
...
-					Required: true,
-					Usage:    "repository to download",
+					Name:  "r",
+					Usage: "repository url",
 				},
 			},
 			Action: func(c *cli.Context) error {
-				CloneTemplate(c)
+				if c.NumFlags() == 0 {
+					// ValidateProject()
```
Comment (jcockbain96) [inline actions/commands.go]: Could make this `TODO: Add ValidateProject() function`  to be super clear what is happening (if this is being merged as-is)?

File: actions/commands.go [hunk 3]
```diff
@@ -35,33 +35,20 @@ func Commands() {
 	app.Commands = []cli.Command{
 
 		{
-			Name:    "clone",
-			Aliases: []string{"c"},
-			Usage:   "Clone a template from github",
+			Name:  "project",
+			Usage: "Project management",
```
Comment (jcockbain96) [inline actions/commands.go]: Something more like `Manage Codewind projects` for now? Know this will probably change as we develop this!

**Comment by hhellyer** [comment]
Running a build here: https://ci.eclipse.org/codewind/job/Codewind/job/codewind-installer/view/change-requests/job/PR-64

**Comment by hhellyer** [comment]
We can merge this now as we are still waiting for the validation code.
Once that is in the only output should be something like:

`{"status":"success","result":{"language":"unknown","projectType":"docker"}}`

Currently there is too much logging coming out. However we can remove that when we add the validation code and return the JSON describing the project.

**Pull request merged**